File: com.test/src/easy/ReverseLinkedList.java
package easy;

import common.ListNode;

public class ReverseLinkedList {
    public ListNode reverseList(ListNode head) {
        ListNode temp;
        ListNode current = head;
        ListNode previous = null;
        while (current != null) {
            temp = current.next;
            current.next = previous;
            previous = current;
            current = temp;
        }
        return previous;
    }
}

File: go/linked-lists/reverse-linked-list.go
package main

func reverseList(head *ListNode) *ListNode {
	if head == nil {
		return head
	}

	left := head
	right := head.Next

	head.Next = nil

	var cur *ListNode

	for right != nil {
		cur = left
		left = right
		right = right.Next
		left.Next = cur
	}

	return left
}
